Frends.AmazonS3.ListObjects/Frends.AmazonS3.ListObjects/ListObjects.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.ComponentModel;
+using System.Threading;
+using Frends.AmazonS3.ListObjects.Definitions;
+using System.Threading.Tasks;
+using Amazon;
+using Amazon.S3;
+using System.Collections.Generic;
+using Amazon.S3.Model;
+
+namespace Frends.AmazonS3.ListObjects
+{
+    /// <summary>
+    /// Amazon S3 task.
+    /// </summary>
+    public class AmazonS3
+    {
+        /// <summary>
+        /// List objects in AWS S3.
+        /// </summary>
+        /// <param name="source">Source connection parameters</param>
+        /// <param name="options">Options for the task</param>
+        /// <param name="cancellationToken">Token to stop ListObjects. This is generated by Frends.</param>
+        /// <returns>object { string BucketName, string Key, string Etag, long Size, DateTime LastModified  }</returns>
+        public static async Task<Result> ListObjects([PropertyTab] Source source, [PropertyTab] Options options, CancellationToken cancellationToken)
+        {
+            var response = await ListBucketContentsAsync(source, options, cancellationToken);
```

Comment:
I don't think it is a good idea to just move the whole task functionality into helper method and then just call that helper method. You should execute the task steps in the task method.